### ISSUE TITLE
[ONNX export] Support image normalization in graph

### ISSUE THREAD

**Issue by francescorubbo** [open-mmlab/mmdetection#5958]
**Describe the feature**
Add option to include the image normalization preprocessing step in the ONNX graph.

**Motivation**
In order to export models to truly portable ONNX files, the preprocessing steps need to be included in the graph as much as possible. There is also a computational efficiency aspect, as normalization would happen on the GPU (e.g. see https://github.com/open-mmlab/mmdetection/issues/4488)

**Related resources**
A potential implementation would consist in passing the `normalize_cfg` to the `forward` method so that it can be used to normalize the image. I.e. adding the block below at https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/detectors/base.py#L169
```
            onnx_input = img[0]
            normalize_cfg = kwargs.get('normalize_cfg', None)
            if normalize_cfg:
                mean = torch.tensor(normalize_cfg['mean'])[..., None, None]
                std = torch.tensor(normalize_cfg['std'])[..., None, None]
                onnx_input = (img[0] - mean) / std
            return self.onnx_export(onnx_input, img_metas[0])
```

Then `tools/deployment/pytorch2onnx.py` can be modified accordingly to support the option.

**Additional context**
If approved, I'd be happy to open a PR with the changes above (and/or alternative/additional suggestions).

**Comment by jshilong** [comment]
Thanks for your suggestion!
Actually, we can pass `img_norm_cfg` via `img_meta` which is consistent with model training and testing.
We would appreciate it if you can create a PR to add this feature, but you need to add this feature as optional to our own export script to avoid bc-breaking.
